taskchampion/taskchampion/src/errors.rs

```diff
@@ -12,4 +16,23 @@ pub enum Error {
     /// other irrecoverable error.
     #[error("Local replica is out of sync with the server")]
     OutOfSync,
+    /// A usage error
+    #[error("User Error: {0}")]
+    UserError(String),
+
+    /// Error conversions.
+    #[error(transparent)]
+    Http(#[from] ureq::Error),
+    #[error(transparent)]
+    Io(#[from] io::Error),
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+    #[error(transparent)]
+    Other(#[from] anyhow::Error),
+    #[error("Third Party Sqlite Error")]
+    Rusqlite(#[from] rusqlite::Error),
+    #[error(transparent)]
+    Sqlite(#[from] crate::storage::sqlite::SqliteError),
 }
+
+pub type Result<T> = core::result::Result<T, Error>;
```

Comment:
Since this crate uses `std` all over the place, you can refer to this as `std::result::Result`.  (The `std` crate just re-exports everything in `core`, so they're precisely the same type)